moped-editor/src/views/projects/projectView/ProjectComponents.js

```diff
@@ -71,29 +72,30 @@ const ProjectComponents = () => {
     },
   });
 
-  // Return loading if not in progress
-  if (loading) return <CircularProgress />;
-
-  if (error) return <div>{JSON.stringify(error)}</div>;
-
   /**
    * Retrieve and flatten a nested list of features:
    *  moped_proj_components -> moped_proj_features_components -> moped_proj_feature
    */
-  const projectFeatureRecords = data.moped_proj_components.reduce(
-    (accumulator, component) => [
-      ...accumulator,
-      // Append if current component is selected, or none are selected (0)
-      ...(selectedComp === component.project_component_id || selectedComp === 0
-        ? component.moped_proj_features_components.map(featureComponent => ({
-            ...featureComponent.moped_proj_feature,
-            project_features_components_id:
-              featureComponent.project_features_components_id,
-          }))
-        : []),
-    ],
-    []
-  );
+  const projectFeatureRecords =
+    data && data?.moped_proj_components
+      ? data.moped_proj_components.reduce(
+          (accumulator, component) => [
+            ...accumulator,
+            // Append if current component is selected, or none are selected (0)
+            ...(selectedComp === component.project_component_id ||
+            selectedComp === 0
+              ? component.moped_proj_features_components.map(
+                  featureComponent => ({
+                    ...featureComponent.moped_proj_feature,
+                    project_features_components_id:
+                      featureComponent.project_features_components_id,
+                  })
+                )
+              : []),
+          ],
+          []
+        )
+      : [];
```

Comment:
This part looks scarier, but the only thing I did was to add the line `data && data?.moped_proj_components`, this is because it was no longer protected by `if(loading)` or `if(error)` which had to be moved down because useEffect cannot be used conditionally.